homeassistant/components/peco/sensor.py

```diff
@@ -108,6 +103,6 @@ def __init__(
         self.entity_description = description
 
     @property
-    def native_value(self) -> float:
+    def native_value(self) -> int:
         """Return the value of the sensor."""
-        return self.coordinator.data[self.entity_description.key]
+        return int(getattr(self.coordinator.data, self.entity_description.key))
```

Comment:
Why is `getattr` introduced here?

Reply:
I changed to using pydantic on the upstream library, which sets the variables as attributes and not something subscriptable

Reply:
Right, in that case, please add a lambda to the entity description, so typing can be kept as well (see for example, the `tailscale` integration, which does use pydantic as well, and uses a `state_fn` in the entity description to hold a lambda).